(docs) Add documentation for upcoming Background Service restrictions (#1246)
chocolatey/docs: src/content/docs/en-us/features/self-service-anywhere.mdx

2. Are focused on package management (not system configuration).
3. Have well-defined behavior in non-interactive contexts.

are allowed to run through the Background Service. This ensures a secure, predictable experience for both users and administrators.

#### Allowed Commands

The operational commands that are allowed to run through the Background Service are:

* `cache` - Manages cached HTTP responses from configured sources to prevent redundant server lookups.
* `download` - Downloads packages without installing them.
* `install` - Installs software packages from approved sources.
* `new` - Generates new package specification files from templates.
* `optimize` - Optimizes installed packages by reducing cache, removing installers, and minimizing disk space usage.
* `pin` - Adds or removes package pins to control whether packages can be upgraded or uninstalled.
* `sync` - Creates packages based on software listed in Programs and Features if a package for that software hasn't already been installed.
* `uninstall` - Removes installed software packages.
* `upgrade` - Updates packages to their latest available versions.

<Callout type="info">

>     By default, Chocolatey CLI configures `backgroundServiceAllowedCommands` to `"install,upgrade"` out of the box.

Reviewer: ```suggestion
    By default, Chocolatey CLI sets `backgroundServiceAllowedCommands` to `"install,upgrade"`.
```

</Callout>

#### Common Configuration Examples

Most organizations configure a subset of these commands based on their self-service needs:

**Basic self-service** (install and upgrade only):
```powershell

> choco config set backgroundServiceAllowedCommands "install,upgrade"

Reviewer: ```suggestion
choco config set --name backgroundServiceAllowedCommands --value "'install,upgrade'"
```

Author: I have created the issue https://github.com/chocolatey/choco/issues/3824 to keep track of getting this updated in the commandline help documentation.
